exercises/frontend/src/app/page.tsx

```diff
@@ -63,21 +65,17 @@ export const Home = () => {
 const NameEditor = (props: {
   fetchedFirstName: string | null,
   fetchedLastName: string | null,
+  fullName: string | null,
   onSaveNames: (firstName: string, lastName: string) => void,
 }) => {
   const {
     fetchedFirstName,
     fetchedLastName,
+    fullName,
```

Comment:
nit: wouldn't it be cleaner to calculate fullName inside NameEditor? So that you don't need to pass in an extra prop (fullName) that can be calculated from other props

Reply:
I tried to limit the functionality of the `NameEditor` such that all it does is trigger the `firstName` and `lastName` change and it remains a dumb component otherwise.
You think it makes more sense to have it calculated as close as possible to where it is rendered?

Reply:
Hmm.. yeah I think so. Unless the `fullName` needs to be passed elsewhere (right now it doesn't), I think computing it as close to where it's calculated makes the most sense.

One way to think about it is if we were testing this component, technically it's possible for `fullName` to be inconsistent with `fetchedFirstName` and `fetchedLastName` (there's no type validation to prevent this). This is weird state of affairs, so we should avoid it where possible by removing unnecessary props.